packages/json-api/src/Registry.ts

```diff
@@ -0,0 +1,92 @@
+import { JSONValue, JSONAPIDocument, JSONAPIResource } from "./jsonApiTypes";
+import {
+  IncludesMap,
+  MapObject,
+  Registry,
+  Resource,
+  ResourceParseOptions,
+  ResourceSpec
+} from "./types";
+import { identity, kebabCaseDeep, camelCaseDeep, TransformFunc } from "./utils";
+import ResourceImpl from "./Resource";
+
+export type RegistryCreateOptions = {
+  keyTransform?: "kebab";
+};
+
+export default class RegistryImpl implements Registry {
+  private readonly options?: RegistryCreateOptions;
+
+  private readonly resources: MapObject<Resource>;
+
+  public readonly keyTransformFunc: TransformFunc;
+
+  public readonly keyParseFunc: TransformFunc;
+
+  constructor(options?: RegistryCreateOptions) {
+    this.options = options;
+    this.resources = {};
+
+    this.keyTransformFunc =
+      this.options?.keyTransform === "kebab" ? kebabCaseDeep : identity;
+    this.keyParseFunc =
+      this.options?.keyTransform === "kebab" ? camelCaseDeep : identity;
+  }
+
+  define(type: string, spec: ResourceSpec): Resource {
+    const resource = new ResourceImpl(this, type, spec);
+    if (type in this.resources) {
+      throw new Error(`Resource ${type} is already defined`);
+    }
+    this.resources[type] = resource;
+    return resource;
+  }
+
+  find(type: string): Resource | null {
+    return this.resources[type];
+  }
+
+  format(type: string, data: JSONValue): JSONAPIDocument {
+    const resource = this.find(type);
+    if (!resource) {
+      throw new Error(`Cannot find resource ${type}`);
+    }
+    return resource.document(data);
+  }
```

Comment:
```suggestion
  format(type: string, data: JSONValue): JSONAPIDocument {
    const resource = this.find(type);
    return resource.document(data);
  }
```

Reply:
As above - find() can return null if the resource is not found, whereas format() cannot, and therefore throws an exception.